pkg/fs/fs_test.go

```diff
@@ -253,6 +253,44 @@ func TestReadFileFromRoot(t *testing.T) {
 	})
 }
 
+func TestTruncateFile(t *testing.T) {
+	condSkip(t)
+	inEmptyMutDir(t, func(env *mountEnv, rootDir string) {
+		tmpFile, err := ioutil.TempFile(rootDir, "camlitest")
+		if err != nil {
+			t.Fatal(err)
+		}
+		if _, err = tmpFile.WriteString("hello world from test"); err != nil {
+			t.Fatal(err)
+		}
+		if err = tmpFile.Close(); err != nil {
+			t.Fatal(err)
+		}
+
+		tmpFile, err = os.OpenFile(tmpFile.Name(), os.O_RDWR, 0644)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if err = tmpFile.Truncate(6); err != nil {
+			t.Fatal(err)
+		}
+		if _, err = tmpFile.WriteAt([]byte("perkeep"), 6); err != nil {
+			t.Fatal(err)
+		}
+		if err = tmpFile.Close(); err != nil {
+			t.Fatal(err)
+		}
+
+		contents, err := ioutil.ReadFile(tmpFile.Name())
+		if err != nil {
+			t.Fatal(err)
+		}
+		if cont := string(contents); cont != "hello perkeep" {
+			t.Fatalf("expected file content to be 'hello perkeep', got '%s'", cont)
```

Comment:
got %q, not '%s'

but in Go, the "got" usually comes first, and then the want.

    if want := "hello perkeep"; string(got) != want {
            t.Fatalf("file contents = %q; want %q", got, want)
    }

Btw, if you do the `string([]byte)` conversion later, in arguments to `==` or `!=`, it doesn't end up allocating. (Not that it matters in this case)

Reply:
good information about the allocations, i did not know that, thanks! had to look up '%q' too, i have to admit! I updated everything to your comments in the latest version!